packages/start/src/config/index.ts

```diff
@@ -16,64 +16,55 @@ import { serverFunctions } from '@vinxi/server-functions/plugin'
 // @ts-expect-error
 import { serverTransform } from '@vinxi/server-functions/server'
 import { z } from 'zod'
+import type { RouterSchemaInput } from 'vinxi'
 import type { Manifest } from '@tanstack/react-router'
 import type * as vite from 'vite'
 
-const viteSchema = z
-  .object({
-    plugins: z.function().returns(z.array(z.custom<vite.Plugin>())).optional(),
-  })
-  .optional()
-  .default({})
-
-const babelSchema = z
-  .object({
-    plugins: z
-      .array(z.union([z.tuple([z.string(), z.any()]), z.string()]))
-      .optional(),
-  })
-  .optional()
-  .default({})
-
-const reactSchema = z
-  .object({
-    babel: babelSchema,
-  })
-  .optional()
-  .default({})
-
-const routersSchema = z
-  .object({
-    ssr: z
-      .object({
-        entry: z.string().default('./app/ssr.tsx'),
-        vite: viteSchema,
-      })
-      .optional()
-      .default({}),
-    rsc: z
-      .object({
-        vite: viteSchema,
-      })
-      .optional()
-      .default({}),
-    client: z
-      .object({
-        entry: z.string().optional().default('./app/client.tsx'),
-        base: z.string().optional(),
-        vite: viteSchema,
-      })
-      .optional()
-      .default({}),
-    server: z
-      .object({
-        vite: viteSchema,
-      })
-      .optional()
-      .default({}),
-  })
-  .optional()
-  .default({})
+const viteSchema = z.object({
+  plugins: z.function().returns(z.array(z.custom<vite.Plugin>())).optional(),
```

Comment:
I believe the `vite.Plugin` type is currently being inherited from the monorepo and not the actual `start` package so you type mismatches like in the `start-basic` example.

Also, do we need to be `zod` validating this? Maybe this could just be a type we pass into the schemas down the line and call `.default({})` on em?

Reply:
It's not really validating the runtime value inside of the array, just providing the type information through the schema. Should be fine.